official/wasm/templates/wasm/wasm.go

```diff
@@ -0,0 +1,235 @@
+		if err != nil {
+			return err
+		}
+
+		// Import
+		template := `wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+%[1]v`
+		replacement := fmt.Sprintf(template, module.PlaceholderSgAppModuleImport)
+		content := replacer.Replace(fConfig.String(), module.PlaceholderSgAppModuleImport, replacement)
+
+		// Init genesis / begin block / end block
+		template = `wasmtypes.ModuleName,
+%[1]v`
+		replacement = fmt.Sprintf(template, module.PlaceholderSgAppInitGenesis)
+		content = replacer.Replace(content, module.PlaceholderSgAppInitGenesis, replacement)
+		replacement = fmt.Sprintf(template, module.PlaceholderSgAppBeginBlockers)
+		content = replacer.Replace(content, module.PlaceholderSgAppBeginBlockers, replacement)
+		replacement = fmt.Sprintf(template, module.PlaceholderSgAppEndBlockers)
+		content = replacer.Replace(content, module.PlaceholderSgAppEndBlockers, replacement)
+
+		// Mac Perms
+		template = `{Account: wasmtypes.ModuleName, Permissions: []string{authtypes.Burner}},
+%[1]v`
+		replacement = fmt.Sprintf(template, module.PlaceholderSgAppMaccPerms)
+		content = replacer.Replace(content, module.PlaceholderSgAppMaccPerms, replacement)
+
+		return r.File(genny.NewFileS(configPath, content))
+	}
+}
+
+// appModify app.go modification when adding wasm integration.
+func appModify(replacer placeholder.Replacer, opts *Options) genny.RunFn {
+	return func(r *genny.Runner) error {
+		appPath := filepath.Join(opts.AppPath, module.PathAppGo)
+		f, err := r.Disk.Find(appPath)
+		if err != nil {
+			return err
+		}
+
+		// Import
+		template := `wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
+%[1]v`
+		replacement := fmt.Sprintf(template, module.PlaceholderSgAppModuleImport)
+		content := replacer.Replace(f.String(), module.PlaceholderSgAppModuleImport, replacement)
+
+		// Keeper declaration
+		template = `
+// CosmWasm
+WasmKeeper       wasmkeeper.Keeper
+ScopedWasmKeeper capabilitykeeper.ScopedKeeper
+
+%[1]v`
+		replacement = fmt.Sprintf(template, module.PlaceholderSgAppKeeperDeclaration)
+		content = replacer.Replace(content, module.PlaceholderSgAppKeeperDeclaration, replacement)
+
+		content, err = goanalysis.ReplaceReturn(
+			content,
+			"New",
+			"app",
+			"app.WasmKeeper.InitializePinnedCodes(app.NewUncachedContext(true, tmproto.Header{}))",
+		)
+		if err != nil {
+			return err
+		}
+
+		return r.File(genny.NewFileS(appPath, content))
+	}
+}
+
+// ibcModify ibc.go modification when adding wasm integration.
+func ibcModify(replacer placeholder.Replacer, opts *Options) genny.RunFn {
+	return func(r *genny.Runner) error {
+		ibcPath := filepath.Join(opts.AppPath, "app/ibc.go")
+		f, err := r.Disk.Find(ibcPath)
+		if err != nil {
+			return err
+		}
+
+		if !strings.Contains(f.String(), "registerIBCModules(appOpts servertypes.AppOptions) error") {
+			return fmt.Errorf("chain does not support wasm integration. See the ignite migration guide")
```

Comment:
Maybe the error could indicate that the app was not scaffolded not migrated to support Ignite CLI >= `v28` and Cosmos SDK `v0.50`